eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageTypes.scala

```diff
@@ -222,6 +222,11 @@ case class ChannelUpdate(signature: ByteVector64,
   require(((messageFlags & 1) != 0) == htlcMaximumMsat.isDefined, "htlcMaximumMsat is not consistent with messageFlags")
 }
 
+case class OnionPacket(version: Int,
+                       publicKey: ByteVector,
```

Comment:
Because I need to return a specific error if that cannot be parsed as a public key (spec compliance) and I think this logic shouldn't bleed in the encoding layer here (that error is returned in Sphinx.scala when we convert to a public key).

Reply:
Isn't that an argument for a tighter coupling between the onion and the codec?

Reply:
I don't know, for crypto stuff I think it makes sense that the wire layer splits the different fields but doesn't interpret them, and lets another layer interpret the fields...

Reply:
Discussed with @sstone, in this particular case it is easier to not type the field.